### DISCUSSION TITLE
Unable To Run Any Languages via Local Instance

### DISCUSSION THREAD

**Discussion by JasonSteving99** [raxod502/riju#139] [Q&A] [answered]
Hi @raxod502 ! This project is awesome and I'm super excited to try to get my hobby language [Claro](https://clarolang.com) included into it if at all possible so I've been trying to get started by running a Riju instance locally but so far no luck. 

I'm following the instructions at https://github.com/raxod502/riju/blob/master/doc/local.md, and I'm able to access the home page via http://localhost:6119/, however, when I click into any language, there is an essentially empty page like the following:

<img width="1680" alt="Screen Shot 2021-11-09 at 11 34 59 AM" src="https://user-images.githubusercontent.com/32336750/140992526-6bf83480-9128-459b-a39a-4fe63085bf07.png">

The "Run" button doesn't do anything, and the editor on the left isn't interactive. Is this expected?

This may be unrelated, but I'll note that I actually am seeing the following error when running `make shell I=runtime E=1 CMD="make dev"` 

<img width="1055" alt="Screen Shot 2021-11-09 at 11 36 54 AM" src="https://user-images.githubusercontent.com/32336750/140992642-8aa3322d-71cb-4a4d-a886-7136b8137115.png">

Note that it says `Loaded 2 language configuration(s) from disk` because to try to simplify my life, I removed all `langs/*.yaml` files except for `langs/python.yaml` and `langs/claro.yaml`

I don't think that this is the cause of the issue because I'm still seeing the home page load these two just fine:

<img width="1680" alt="Screen Shot 2021-11-09 at 11 40 28 AM" src="https://user-images.githubusercontent.com/32336750/140993224-ec8aeb57-3346-4ae7-ac76-4007b662f516.png">

Do you have any hints on how I can continue making progress towards onboarding Claro to Riju? If it's useful info, here's the current state of my `langs/claro.yaml` https://github.com/JasonSteving99/riju/blob/master/langs/claro.yaml#L12-L20 

Thanks in advance

**Comment by raxod502** [comment, marked as answer]
Hey! Sorry it's difficult to get things set up locally. I have some plans of rearchitecting things so that it's less of a hassle, but I'll be happy to help you get things going as they stand.

If Webpack is failing then probably no JavaScript is compiled. Have you checked the browser console? It probably says it has failed to load the required scripts.

As for why Webpack is failing, it looks like this may be an upstream issue: https://github.com/webpack/webpack/issues/14532. The fix was merged in https://github.com/webpack/webpack/pull/14584, but I don't think it's been backported to Webpack 4, which we're stuck on because upgrading is nontrivial to say the least (https://github.com/raxod502/riju/issues/61).

So, in the meantime, let's try the safest workaround suggested in the thread, which is to downgrade to Node 16. I've pushed updates to the admin and runtime Dockerfiles in https://github.com/raxod502/riju/commit/cf47e68727682e2519b407268ea390e2dade9489, so if you rebuild I think this error should be gone. Let me know.

**Reply by JasonSteving99** [reply]
You 100% solved my problems, thank you!

<img width="1680" alt="Claro Running on Riju!" src="https://user-images.githubusercontent.com/32336750/142338044-ef94626c-eac7-4ead-adad-d62d4d3a5483.png">

I'm super excited to get this merged back into Riju, but I have a few more things left to iron out before it'd be acceptable. Currently on first time clicking "Run", this proof of concept actually rebuilds the entire Claro compiler from source along with compiling the user's program itself...lol. I'll get back to some changes on my end and then submit a pull request. Super excited about this progress though, thanks again!
